inc/packages/namespace.php

```diff
@@ -0,0 +1,582 @@
+<?php
+/**
+ * Altis Search Packages.
+ *
+ * @package altis/search
+ */
+
+namespace Altis\Enhanced_Search\Packages;
+
+use Altis\Enhanced_Search;
+use ElasticPress\Elasticsearch;
+use ElasticPress\Indexables;
+use ElasticPress\Utils;
+use WP_Error;
+
+/**
+ * Bind hooks for ElasticSearch Packages.
+ *
+ * @return void
+ */
+function setup() {
+	add_action( 'admin_enqueue_scripts', __NAMESPACE__ . '\\enqueue_scripts' );
+	add_action( 'admin_init', __NAMESPACE__ . '\\handle_form' );
+	add_action( 'admin_menu', __NAMESPACE__ . '\\admin_menu', 5 );
+	add_action( 'network_admin_menu', __NAMESPACE__ . '\\admin_menu' );
+	add_filter( 'removable_query_args', __NAMESPACE__ . '\\add_removable_query_args' );
+
+	// Background tasks.
+	add_action( 'altis.search.updated_packages', __NAMESPACE__ . '\\on_updated_packages', 10, 2 );
+	add_action( 'altis.search.update_index_settings', __NAMESPACE__ . '\\do_settings_update', 10, 2 );
+}
+
+/**
+ * Add network admin page.
+ *
+ * @return void
+ */
+function admin_menu() {
+	add_menu_page(
+		__( 'Search Configuration', 'altis' ),
+		__( 'Search Config', 'altis' ),
+		is_network_admin() ? 'manage_network_options' : 'manage_options',
+		'search-config',
+		__NAMESPACE__ . '\\admin_page',
+		'dashicons-search',
+		171
+	);
+}
+
+/**
+ * Add our custom success/error messaging query args to the removable list.
+ *
+ * @param array $removable_query_args Array of query parameter names to remove from the URL in the admin.
+ * @return array
+ */
+function add_removable_query_args( array $removable_query_args ) : array {
+	$removable_query_args[] = 'did_update';
+	return $removable_query_args;
+}
+
+/**
+ * Enqueue scripts and styles for the search config form.
+ *
+ * @param string $hook_suffix The admin page hook.
+ * @return void
+ */
+function enqueue_scripts( string $hook_suffix ) {
+	if ( $hook_suffix !== 'toplevel_page_search-config' ) {
+		return;
+	}
+
+	wp_enqueue_style( 'wp-components' );
+}
+
+/**
+ * Includes the search config page template.
+ *
+ * @return void
+ */
+function admin_page() {
+	$types = [ 'synonyms', 'stopwords', 'user_dictionary' ];
+
+	$for_network = is_network_admin();
+
+	foreach ( $types as $type ) {
+		$uploaded_file_var = "{$type}_uploaded_file";
+		$manual_file_var = "{$type}_manual_file";
+		$text_var = "{$type}_text";
+		$file_date_var = "{$type}_file_date";
+		$uploaded_file_status = "{$type}_uploaded_status";
+		$manual_file_status = "{$type}_manual_status";
+		$uploaded_file_error = "{$type}_uploaded_error";
+		$manual_file_error = "{$type}_manual_error";
+
+		$file_type_string = str_replace( '_', '-', $type );
+
+		if ( $for_network ) {
+			$$uploaded_file_status = get_site_option( "altis_search_package_status_uploaded-{$file_type_string}" );
```

Comment:
Are these intentionally variable variables? Can we use an array or something instead to make this more explicit? Variable variables are very easy to miss while reading the code.

(If it's for ergonomics for the template file, then an `extract()` of the data later would be better, but the arrays are probably fine anyway.)

Sidenote, I don't love how variable these option names are, and they might also start getting close to the maximum option name length?

Reply:
Yeah I just threw the page together fairly quickly and it went through a lot of iteration in that time. I'm not really into using `extract()` so I think an array will be better as you say.

I did worry about max option name length but for the hard coded values currently in use we're ok.